src/services/stringCompletions.ts

```diff
@@ -277,13 +277,14 @@ namespace ts.Completions.StringCompletions {
         return nameAndKind(name, ScriptElementKind.directory, /*extension*/ undefined);
     }
 
-    function addReplacementSpans(text: string, textStart: number, names: readonly NameAndKind[]): readonly PathCompletion[] {
+    function addReplacementSpans(text: string, textStart: number, names: readonly NameAndKind[], tsConfigPaths?: string[]): readonly PathCompletion[] {
         const span = getDirectoryFragmentTextSpan(text, textStart);
-        return names.map(({ name, kind, extension }): PathCompletion => ({ name, kind, extension, span }));
+        const pathSpan = getWholeFragmentTextSpan(text, textStart);
+        return names.map(({ name, kind, extension }): PathCompletion => tsConfigPaths?.some(path => path === name) ? { name, kind, extension, span: pathSpan } : { name, kind, extension, span });

```

Comment:
```suggestion
        return names.map(({ name, kind, extension }): PathCompletion => contains(tsConfigPaths, name) ? { name, kind, extension, span: pathSpan } : { name, kind, extension, span });
```

Reply:
Would it also work just to say “if `name` contains a path separator, assume it’s a completion for the whole path, so replace everything”? That way if we introduce path completions for other full paths in the future, we don’t hit the same bug?

Reply:
!
Good point!